### DISCUSSION TITLE
Regex Manager - Registry URL Template

### DISCUSSION THREAD

**Discussion by 56KBs** [renovatebot/renovate#8589] [Request Help] [answered]
Hi,

I'm looking to run the regex manager on a resource which is hosted in a private GitLab instance. Is it possible to configure the registryUrl without using a capture group?

Here is some example configuration:

```json
    "regexManagers": [
      {
        "fileMatch": ["(^|/)Jenkinsfile$"],
        "matchStrings": [
          "@Library\\(\"jenkinsLib@(?<currentValue>.*?)\"\\)_"
        ],
        "datasourceTemplate": "gitlab-tags",
        "depNameTemplate": "group/repositoryName",
        "versioningTemplate": "semver",
      }
    ],
```

My issue is that `group/repositoryName` is hosted on `https://gitlab.my-private-instance.com`, meaning I need to configure a custom registry URL.

The configuration suggests this is possible using the registryUrl capture group, however in my scenario, I don't have text to match on that details the registry this resource is hosted on. I understand I could expand my regex to include the additional match and add the registry URL within my Jenkinsfile, however I am attempting to get this to function on a number of repositories without me having to update all of the repositories to include this additional information (Which may change in future too).

Is there any way to do what I'm looking to do, or would this be an enhancement to renovate?

Edit: From looking at how the code functions, it appears this functionality could be added via using a `registryUrlTemplate` configuration field which translates to the `registryUrls` as a single length array (Similar to the capture group itself). This obviously requires code changes to complete, so I was hoping there may be a way to do it without code changes - Although I'd happily contribute this functionality.

**Comment by rarkins** [comment, marked as answer]
I think it makes most sense to allow `registryUrlTemplate`, which shouldn't be much work

**Reply by fgreinacher** [reply]
FYI: This is being implemented in https://github.com/renovatebot/renovate/pull/8611 and https://github.com/renovatebot/renovate/pull/8819.
